woocommerce/woocommerce-ios
# Centralize image downloading/caching service and show image in Product description Aztec editor

Fixes #1500 

## Changes

- Created an `ImageService` protocol with all the image download and caching services. Implemented `ImageService` with `DefaultImageService` using Kingfisher
- Added `ImageService` to `ServiceLocator` for the whole app to access
- Moved `UIImageView` extension for `setImage` with Kingfisher to `DefaultImageService` implementation, so that we use the same image cache for all the image services
  - Replaced the existing `UIImageView` extension usage with the new `ImageService` call, and DI'ed `ImageService` when needed
- In `AztecTextViewAttachmentHandler`, retrieved the image from the cache and then downloaded the image
- Cleared the image cache when the user logs out
- Added unit tests

## Testing

Prerequisite: the store has a simple product with at least one image in the description

- Launch the app
- Go to Orders tab, and tap on a few orders --> the product image should show correctly in the Order Details, and the image is shown immediately after the first download
- Go to Products tab --> the product image should show correctly in the product cell, and the image is shown immediately after the first download
- Tap on a few products --> the product images in the header should show correctly in the product cell, and the image is shown immediately after the first download
- Tap on a simple product with an image in the description
- Tap "Edit" in the navigation bar
- Tap on the description row --> the image(s) should show up in the editor
- Log out from the app
- Log in again
- Repeat the steps before logging out --> the images should be downloaded again (starting from placeholder) then cached

(if interested, can also check the files under simulator or device's `AppData > Library > Caches > com.onevcat.Kingfisher.ImageCache.default`)

Update release notes:
